cuml/rf_demo.ipynb

```diff
@@ -0,0 +1,439 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "(10500000, 28) (10500000,) (500000, 28) (500000,)\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(X_train.shape, y_train.shape, X_test.shape, y_test.shape)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# cuml Random Forest params\n",
+    "\n",
+    "cu_rf_params = {\n",
+    "    'n_estimators': 25,\n",
+    "    'max_depth': 13,\n",
+    "    'n_bins': 15,\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 40.4 s, sys: 13.1 s, total: 53.5 s\n",
+      "Wall time: 53.6 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "# Train cuml RF\n",
+    "\n",
+    "cu_rf = cuRF(**cu_rf_params)\n",
+    "cu_rf.fit(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# sklearn Random Forest params\n",
+    "\n",
+    "skl_rf_params = {\n",
+    "    'n_estimators': 25,\n",
+    "    'max_depth': 13,\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 56min 27s, sys: 1min, total: 57min 27s\n",
+      "Wall time: 4min 30s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "# Train sklearn RF parallely\n",
+    "\n",
+    "skl_rf = sklRF(**skl_rf_params, n_jobs=20)\n",
+    "skl_rf.fit(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "cuml RF Accuracy Score:  0.72466\n",
+      "sklearn RF Accuracy Score:  0.72177\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Predict\n",
+    "print(\"cuml RF Accuracy Score: \", accuracy_score(cu_rf.predict(X_test), y_test))\n",
+    "print(\"sklearn RF Accuracy Score: \", accuracy_score(skl_rf.predict(X_test), y_test))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.7.3"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 2
+}
```

Comment:
Are you planning to add the hyper-parameter search to this notebook? Or, planning to create a separate notebook for that?

Reply:
Ideally, a separate notebook. That's because I can't use sklearn's `GridSearchCV` with our RF implementation yet, and I believe adding trivial for loops for hyperparam sweep will make the code look less readable for a demo notebook.